scripts/src/blockchain/transactionBuilder.ts

```diff
@@ -25,8 +34,8 @@ export class TransactionBuilder {
 		return this;
 	}
 
-	public addMemo(memo: string): this {
-		(this as any).memo = memo;
+	public addMemo(memo: Memo<MemoType.Text> | undefined, appId: string): this {
```

Comment:
the public API of memo shouldn't contain `appId` (developer shouldn't add this param, it will be propagated through the kin account it was created on).
Also, we should expose the entire options of `Memo` i.e. let the dev choose between id/hash/text memo (to provide a general builder), maybe add a dedicated for text memo that accepts `string`.